### ISSUE TITLE
Allow compatible pip releases

### ISSUE THREAD

**Issue by lordvlad** [hadolint/hadolint#216]
- [x] This is a bug report
- [ ] This is a feature request
- [x] I searched existing issues before opening this one

### Expected behavior
Dockerfile contains a line like `pip install psycopg2~=2.7.4`, hadolint accepts it happily, since the version is declared and sufficiently precise.
### Actual behavior
Dockerfile contains a line like `pip install psycopg2~=2.7.4`, hadolint prints the error ' Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>`'.
But if the dockerfile contains a line `pip install psycopg2==2.7.*`, hadolint accepts it happily, though that line would allow versions below 2.7.4, which is not my intended behavior 

### Steps to reproduce the behavior

```bash
~/ $ hadolint --version
Haskell Dockerfile Linter v1.7.1-0-gfcaddad
~/ $ hadolint Dockerfile
Dockerfile2:1 DL3006 Always tag the version of an image explicitly
Dockerfile2:3 DL3013 Pin versions in pip. Instead of `pip install <package>` use `pip install <package>==<version>`
```

**Dockerfile** (if relevant)

```Dockerfile
from python

run pip3 install psycopg2~=2.7.14
```

**Additional environment details (OS, stack version, etc.)**
Windows & cygwin

### Additional info to python compatible versions
https://www.python.org/dev/peps/pep-0440/#compatible-release

**Comment by lorenzo** [comment]
I'll take a look  at this
